lib/iex/lib/iex/evaluator.ex

```diff
@@ -82,6 +102,13 @@ defmodule IEx.Evaluator do
     end
   end
 
+  defp find_matched_variables(binding, var_prefix) do
+    for {var_name, _value} <- binding,
```

Comment:
Any expression that defines a variable inside a quote:

```elixir
iex(1)> Code.eval_quoted (quote do var = 1 end), []
{1, [{{:var, Elixir}, 1}]}
```

One suggestion is to define a macro inside the test module and invoke that macro on `@tag previous_line: "Mod.macro"`.